pkg/client/client_cache.go

```diff
@@ -82,13 +82,13 @@ func (c *clientCache) getResource(obj runtime.Object) (*resourceMeta, error) {
 	_, isUnstructuredList := obj.(*unstructured.UnstructuredList)
 	isUnstructured = isUnstructured || isUnstructuredList
 
-	// It's better to do creation work twice than to not let multiple
-	// people make requests at once
-	c.mu.RLock()
-	resourceByType := c.structuredResourceByType
+	var resourceByType map[schema.GroupVersionKind]*resourceMeta
 	if isUnstructured {
 		resourceByType = c.unstructuredResourceByType
+	} else {
+		resourceByType = c.structuredResourceByType
 	}
```

Comment:
Field reads don't need to be under the lock as the fields themselves are not mutated, the contents of the maps are mutated. I.e. it's safe to read the map pointer into a variable.